core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -1284,15 +1279,97 @@ public SELF withWorkingDirectory(String workDir) {
         return self();
     }
 
+    private interface CopyFileToContainerWrapper<T> {
```

Comment:
Can we find an interface name with more meaning for this? One might argue, that `Transferable` could have been a good abstraction already (we can transfer bytes of a file into the container), but of course, this is already used differently in our codebase now.

Maybe `CopyFileToContainerSource` ? If this is reflected in some parameters being renamed from `wrapper` to `source`, this should also help readability.